File: MyCalc01.py
print("\n         Welcome To Virtual Calculator :             ")
print("What you want to calculate : 1.Area     2.Perimeter     3.Volume")
choice=int(input("Your choice : "))
if(choice==1):
    print("Area : ")
    print("Choose shape : 1.Triangle    2.Rectangle     3.Circle")
    choice=int(input("Your choice : "))
    if(choice==1):
        print("Triangle : ")
        base = int(input("Enter base of the triangle : "))
        if(base<0):
            print("Invalid input : \n")
        else:
            height = int(input("Enter height of the triangle : "))
            if(height<0):
                print("Invalid input : \n")
            else:
                area = 0.5*base*height
                print ('Area of the triangle : ',area)
    elif(choice==2):
        print("Rectangle : ")
        len = int(input("Enter length of the rectangle : "))
        if(len<0):
            print("Invalid input :\n")
        else:
            breadth = int(input("Enter breadth of the rectangle : "))
            if(breadth<0):
                print("Invalid input : \n")
            else:
                area = len*breadth
                print("Area of the rectangle : ", area)
    elif(choice==3):
        print("Circle : ")
        r = int(input("Enter radius of the circle : "))
        if(r<0):
            print("Invalid input : \n")
        else:
            area = 3.14*r**2
            print("Area of the circle : ", area)
    else:
        print("Invalid input :\n")
elif(choice==2):
    print("Perimeter : ")
    print("Choose shape : 1.Triangle    2.Rectangle     3.Circle")
    choice=int(input("Your choice : "))
    if(choice==1):
        print("Triangle : ")
        base = int(input("Enter base of the triangle : "))
        if(base<0):
            print("Invalid input : \n")
        else:
            height = int(input("Enter height of the triangle : "))
            if(height<0):
                print("Invalid inout : \n")
            else:
                hyp = (base**2+height**2)**0.5
                perim = base+height+hyp
                print("Perimeter of the triangle : ", perim)
    elif(choice==2):
        print("Rectangle : ")
        len=int(input("Enter length of the rectangle : "))
        if(len<0):
            print("Invalid input : \n")
        else:
            breadth = int(input("Enter breadth of the rectangle : "))
            if(breadth<0):
                print("Invalid input : \n")
            else:
                perim = 2*(len+breadth)
                print("Perimeter of the rectangle : ", perim)
    elif(choice==3):
        print("Circle : \n")
        r=int(input("Enter radius of the circle : "))
        if(r<0):
            print("Invalid input : \n")
        else:
            circum = 2*3.14*r
            print("Circumference of the circle : ", circum)
    else:
        print("Invalid input :\n")
elif(choice==3):
    print("\nVolume : ")
    print("Choose shape : 1.Cube    2.Cuboid    3.Cylinder     4.Sphere")
    choice=int(input("Your choice : "))
    if(choice==1):
        print("Cube : ")
        a=int(input("Entre side of the cube : "))
        if(a<0):
            print("Invalid input : \n")
        else:
            vol = a**3
            print("Volume of the cube : ", vol)
    elif(choice==2):
        print("Cuboid : ")
        l=int(input("Enter length of the cuboid : "))
        if(l<0):
            print("Invalid input : \n")
        else:
            b =int(input("Enter breadth of the cuboid : "))
            if(b<0):
                print("Invalid input : \n")
            else:
                h =int(input("Enter height of the cuboid : "))
                if(h<0):
                    print("Invalid input : \n")
                else:
                    vol=l*b*h
                    print("Volume of the cuboid : ", vol)
    elif(choice==3):
        print("Cylinder : 1.Hollow Cylinder     2.Solid Cylinder ")
        choice=int(input("Your choice : "))
        if(choice==1):
            print("Hollow Cylinder : ")
            R=int(input("Enter outer radius : "))
            if(R<0):
                print("Invalid input : \n")
            else:
                r=int(input("Enter inner radius : "))
                if(r<0):
                    print("Invalid input : ")
                else:
                    if(R<r):
                        print("Invalid input : \nOuter radius is smaller than inner radius : \n")
                    else:
                        h=int(input("Enter height of the hollow cylinder : "))
                        if(h<0):
                            print("Invalid input : \n")
                        else:
                            invol=3.14*r**2*h
                            outvol=3.14*R**2*h
                            ratio = outvol/invol
                            print("\nOuter volume : ",outvol,"\nInner volume : ", invol,"\nRatio : ", ratio)
        elif(choice==2):
            print("Solid Cylinder : ")
            r=int(input("Enter radius of the cylinder : "))
            if(r<0):
                print("Invalid input : \n")
            else:
                h=int(input("Enter height of the cylinde : "))
                if(h<0):
                    print("Invalid input : \n")
                else:
                    vol = 3.14*r**2*h
                    print("Volume of the solid cylinder : ",vol)
    elif(choice==4):
        print("Sphere : ")
        print("Choose shape : 1.Hollow Sphere   2.Solid Sphere ")
        choice=int(input("Your choice : "))
        if(choice==1):
            print("Hollow Sphere : ")
            R=int(input("Enter outer radius : "))
            if(R<0):
                print("Invalid input : \n")
            else:
                r=int(input("Enter inner radius : "))
                if(r<0):
                    print("Invalid input : \n")
                else:
                    if(R<r):
                        print("\nInvalid input :\nOuter radius is smaller than inner radius : \n")
                    else:
                        invol = (4/3)*3.14*r**3
                        outvol = (4/3)*3.14*R**3
                        ratio = outvol/invol
                        print("\nOuter volume : ",outvol,"\ninner volume : ",invol,"\nRatio : ",ratio)
        elif(choice==2):
            print("Solid Sphere : ")
            r=int(input("Enter radius of the sphere : "))
            if(r<0):
                print("Invalid input : \n")
            else:
                vol = (4/3)*3.14*r**3
                print("\nVolume of the solid sphere : ", vol)
        else:
            print("Invalid input : \n")
    else:
        print("Invalid input : \n")
else:
    print("Invalid input :\n")

File: MyCalc01.cpp
#include "iostream"
#include "math.h"
using namespace std;
float triangle(int b, int h){
    float area;
    cout<<"\nTriangle : \n";
    cout<<"Enter base : ";
    cin>>b;
    if(b<0){
        cout<<"Invalid input : \n";
    }else{
        cout<<"Enter height : ";
        cin>>h;
        if(h<0){
            cout<<"Invalid input : \n";
        }else{
            area = 0.5*b*h;
            //cout<<"Area = "<<area<<"\n";
            // float circum = b+h;
            // cout<<"Circumference = "<<circum<<"\n";
        }
    }
    return area;
}
float rectangle(int l, int b){
    cout<<"\nRectangle :\n";
    int area;
    cout<<"Enter length : ";
    cin>>l;
    if(l<0){
        cout<<"Invalid input : \n";
    }else{
        cout<<"Enter breadth : ";
        cin>>b;
        if(b<0){
            cout<<"Invalid input : \n";
        }else{
            area = l*b;
            //cout<<"Area = "<<area<<"\n";
            // int circum = l+b;
            // cout<<"Circumference = "<<circum<<"\n";
        }
    }
    return area;
}
float circle(int r){
    cout<<"\nCircle : \n";
    float area;
    cout<<"Enter radius : ";
    cin>>r;
    if(r<0){
        cout<<"Invalid input :\n";
    }else{
        area = 3.14*pow(r, 2);
        //cout<<"Area = "<<area<<"\n";
        // float circum = 2*3.14*r;
        // cout<<"Circumference = "<<circum<<"\n";
    }
    return area;
}
int cube(int a){
    cout<<"Cube : \n";
    int vol;
    cout<<"Enter side : ";
    cin>>a;
    if(a<0){
        cout<<"Invalid input : \n";
    }else{
        vol = pow(a, 3);
        //cout<<"Volume = "<<vol<<"\n";
        // int surface_area = 6*pow(a, 2);
        // cout<<"Surface area = "<<surface_area<<"\n";
        // int circum = 12*a;
        // cout<<"Circumference = "<<circum<<"\n";
    }
    return vol;
}
int cuboid(int l, int b, int h){
    cout<<"Cuboid : \n";
    int vol;
    cout<<"Enter lenght : ";
    cin>>l;
    if(l<0){
        cout<<"Invalid input : \n";
    }else{
        cout<<"Enter breadth : ";
        cin>>b;
        if(b<0){
            cout<<"Invalid input : \n";
        }else{
            cout<<"Enter height : ";
            cin>>h;
            if(h<0){
                cout<<"Invalid input : \n";
            }else{
                vol = l*b*h;
                cout<<"Volume = "<<vol<<"\n";
                int surface_area = 2*(l*b+b*h+l*h);
                cout<<"Surface area = "<<surface_area<<"\n";
                int circum = 4*(l+b+h);
                cout<<"Circumference = "<<circum<<"\n";
            }
        }
    }
    return vol;
}
int main(){
    int choice;
    cout<<"Choose shape : 1.Open shape     2.Close shape\n";
    cout<<"Your choice : ";
    cin>>choice;
    if(choice==1){
        cout<<"Open shape : \n";
        cout<<"Choose shape : 1.Triangle    2.Rectangle    3.Circle\n";
        cout<<"Your choice : ";
        cin>>choice;
        if(choice==1){
            int b, h;
            cout<<"Area of triangle : "<<triangle(b, h);
        }else if(choice==2){
            int l, b;
            cout<<"Area of rectangle : "<<rectangle(l, b);
        }else if(choice==3){
            int r;
            cout<<"Area of circle : "<<circle(r);
        }
    }else if(choice==2){
        cout<<"Close shape : \n";
        cout<<"Choose shape : 1.Cube    2.Cuboid    3.Sphere    4.Cylinder\n";
        cout<<"Your choice : ";
        cin>>choice;
        if(choice==1){
            int a;
            cout<<"Volume of cube : ";
            cout<<cube(a);
        }else if(choice==2){
            int l, b, h;
            cout<<"Volume of cuboid : ";
            cout<<cuboid(l, b, h);
        }
    }else{
        cout<<"Invalid input : \n";
    }
    int op;
    cout<<"\nChoose : 1.Find more :   2.Exit :\n";
    cout<<"Your option : ";
    cin>>op;
    if(op==1){
        main();
    }else if(op==2){
        cout<<"Good bye : \n";
    }else{
        cout<<"Invalid input : \n";
    }
    return 0;
}